### ISSUE TITLE
training loss starts around 2e4

### ISSUE THREAD

**Issue by leinxx** [IDEA-Research/MaskDINO#22]
I tried to train the swin large model on coco, with swin backbone pretrained initilaization. The initial loss starts around 2e4. The loss print looks like this:
```
iter: 19  total_loss: 2.163e+04  loss_ce: 1853  loss_mask: 4.578  loss_dice: 4.78  loss_bbox: 0.05134  loss_giou: 0.03028  loss_ce_dn: 97.37  loss_mask_dn: 4.537  loss_dice_dn: 4.779  loss_bbox_dn: 0.002613  loss_giou_dn: 0.01703  loss_ce_0: 2027  loss_mask_0: 3.692  loss_dice_0: 4.75  loss_bbox_0: 0.07883  loss_giou_0: 0.0301  loss_ce_dn_0: 93.84  loss_mask_dn_0: 4.338  loss_dice_dn_0: 4.748  loss_bbox_dn_0: 0.002613  loss_giou_dn_0: 0.01703  loss_ce_1: 2074  loss_mask_1: 3.531  loss_dice_1: 4.745  loss_bbox_1: 0.05333  loss_giou_1: 0.03028  loss_ce_dn_1: 91.22  loss_mask_dn_1: 3.887  loss_dice_dn_1: 4.74  loss_bbox_dn_1: 0.002613  loss_giou_dn_1: 0.01703  loss_ce_2: 1948  loss_mask_2: 3.681  loss_dice_2: 4.743  loss_bbox_2: 0.05584  loss_giou_2: 0.0301  loss_ce_dn_2: 92.22  loss_mask_dn_2: 4.061  loss_dice_dn_2: 4.766  loss_bbox_dn_2: 0.002613  loss_giou_dn_2: 0.01703  loss_ce_3: 1774  loss_mask_3: 4.15  loss_dice_3: 4.762  loss_bbox_3: 0.05436  loss_giou_3: 0.03028  loss_ce_dn_3: 86.42  loss_mask_dn_3: 4.324  loss_dice_dn_3: 4.754  loss_bbox_dn_3: 0.002613  loss_giou_dn_3: 0.01703  loss_ce_4: 1770  loss_mask_4: 4.513  loss_dice_4: 4.739  loss_bbox_4: 0.05335  loss_giou_4: 0.03519  loss_ce_dn_4: 89.65  loss_mask_dn_4: 4.792  loss_dice_dn_4: 4.726  loss_bbox_dn_4: 0.002613  loss_giou_dn_4: 0.01703  loss_ce_5: 1838  loss_mask_5: 5.272  loss_dice_5: 4.718  loss_bbox_5: 0.05066  loss_giou_5: 0.03384  loss_ce_dn_5: 96.19  loss_mask_dn_5: 5.856  loss_dice_dn_5: 4.696  loss_bbox_dn_5: 0.002613  loss_giou_dn_5: 0.01703  loss_ce_6: 1814  loss_mask_6: 4.328  loss_dice_6: 4.731  loss_bbox_6: 0.04711  loss_giou_6: 0.03028  loss_ce_dn_6: 95.16  loss_mask_dn_6: 4.563  loss_dice_dn_6: 4.727  loss_bbox_dn_6: 0.002613  loss_giou_dn_6: 0.01703  loss_ce_7: 1785  loss_mask_7: 4.494  loss_dice_7: 4.75  loss_bbox_7: 0.0455  loss_giou_7: 0.03191  loss_ce_dn_7: 97.4  loss_mask_dn_7: 4.624  loss_dice_dn_7: 4.761  loss_bbox_dn_7: 0.002613  loss_giou_dn_7: 0.01703  loss_ce_8: 1910  loss_mask_8: 4.013  loss_dice_8: 4.792  loss_bbox_8: 0.04841  loss_giou_8: 0.03385  loss_ce_dn_8: 102.4  loss_mask_dn_8: 4.038  loss_dice_dn_8: 4.791  loss_bbox_dn_8: 0.002613  loss_giou_dn_8: 0.01703  loss_ce_interm: 2027  loss_mask_interm: 3.694  loss_dice_interm: 4.751  loss_bbox_interm: 0.07883  loss_giou_interm: 0.0301  time: 1.6347  data_time: 0.4793  lr: 0.0001  max_mem: 14792M
```

When the network is initialized using your pretrained weights, the initial loss is about 40 some.  This seems to mean that the initialization of weights is not done properly?

**Comment by FengLi-ust** [comment]
Hi, in d2, the checkpointer will print all the loaded and unloaded weights, you can check the output to ensure you load the weights properly.

I also check the loss with my trained log, I also get a similar loss, so your initial loss is reasonable.

**Comment by leinxx** [comment]
Thank you for your reply.  I was able to reduce the initial costs down to a few hundred on my own dataset. The problem is with the `prepare_dn` function. I switched to the prepare_dn function in dino's `dn_components.py`. In my understanding, dino's version implemented contrast dn, while maskdio only implemented dn. It might be a good fit only for my dataset, because my targets are small and non-overlaping. The initial box size can be very wrong and makes matching more noisy?

With the same dataset, batchsize, and lr, on dino, I was able to get `loss_bbox` to 0.036 in 10k iters, and 0.031 at 20k iters.  While in maskdino, `loss_bbox` is 0.06 and 0.05. I don't see any reason to cause the gap, since I am already using the same cdn implementation for both methods. Would you have any suggestions where I should look into?

Thanks

**Comment by FengLi-ust** [comment]
Have you tried the original Mask DINO code? Does it produce similar results?

**Comment by leinxx** [comment]
Yes, I tried the original dino code from its own depo, not fromd detrx. DINO's eval box mIOU converges pretty fast, reached 0.57 at 13k iters, and not improved much after training to 40k iters. MaskDINO achieved 0.38 box IOU at 13 iters. My config for MaskDINO is adopted from `MaskDINO/configs/coco/panoptic-segmentation/swin/maskdino_R50_bs16_50ep_4s_dowsample1_2048.yaml`, with a few modifications:
1. Set COST_MASK_WEIGHT = COST_DICE_WEIGHT = 0 to match without using segmentation masks
2. Reduce DEC_LAYERS from 9 to 6 to match with DINO.
3. Reduce DIM_FEEDFORWARD = 512 in both encoder and decoder. (In DINO, I used the default 2048)
4. dn_noise_scale=0.5, while in DINO, I use dn_box_noise_scale=1 and dn_label_noise_ratio=0.5
5. INITIALIZE_BOX_TYPE='no'

**Comment by FengLi-ust** [comment]
I saw you are using DIM_FEEDFORWARD = 512, which is much smaller than 2048. This may impact the performance a lot.

**Comment by leinxx** [comment]
Possibly. I will need to give this a try. DIM_FEEDFORWARD is reduced mainly to reduce the memory footprint. My dataset only have 3 types of objects of similar sizes, which is much simpler than coco. I feel high dim_feedforward might not be necessary.

**Comment by FengLi-ust** [comment]
dim_feedforward is very important to boost the representation ability of Transformers. You can try using the original dim_feedforward 2048.

**Comment by leinxx** [comment]
![loss](https://user-images.githubusercontent.com/7012985/207822017-d4555518-2961-4802-9cce-d4874e6775e9.png)
The bottom curve is dino's bbox loss, the other two curves are maskdino using dim_feedforward 2048 and 512 separately. Seems that dim_feedforward is not the problem. I also adjusted the noise_scale of box to be consistent with dino.

**Comment by mattiasegu** [comment]
Hi, I'm also finding that the MaskDINO loss starts from very high values (>1e4). Is there a log file that I can perhaps check and compare too? Ideally I would like to know how fast the total loss converges to a reasonable value, and what this reasonable value is. Thanks in advance for your help, and great work @FengLi-ust!

@FengLi-ust @leinxx
